gamestonk_terminal/cryptocurrency/discovery/pycoingecko_model.py

```diff
@@ -0,0 +1,451 @@
+    rows = scrape_gecko_data(url).find("tbody").find_all("tr")
+    results = []
+
+    for row in rows:
+        url = GECKO_BASE_URL + row.find("a")["href"]
+        row_cleaned = clean_row(row)
+        (
+            name,
+            symbol,
+            _,
+            price,
+            *changes,
+            _,
+            _volume,
+            last_added,
+        ) = row_cleaned
+        change_1h, change_24h, _ = changes_parser(changes)
+        results.append([name, symbol, price, change_1h, change_24h, last_added, url])
+    df = replace_qm(pd.DataFrame(results, columns=columns))
+    df.index = df.index + 1
+    df.reset_index(inplace=True)
+    df.rename(columns={"index": "rank"}, inplace=True)
+    return df
+
+
+def get_yield_farms() -> pd.DataFrame:
+    """Scrapes yield farms data from "https://www.coingecko.com/en/yield-farming"
+
+    Returns
+    -------
+    pandas.DataFrame
+        rank, name, pool, audits, collateral,value_locked, returns_year, returns_hour
+    """
+    columns = [
+        COLUMNS["rank"],
+        COLUMNS["name"],
+        "pool",
+        "audits",
+        "collateral",
+        "value_locked",
+        "return_year",
+    ]
+    url = "https://www.coingecko.com/en/yield-farming"
+    rows = scrape_gecko_data(url).find("tbody").find_all("tr")
+    results = []
+    for row in rows:
+        row_cleaned = clean_row(row)[:-2]
+        if " New" in row_cleaned:  # find better way to fix it in future
+            row_cleaned.remove(" New")
+
+        if len(row_cleaned) == 7:
+            row_cleaned.insert(2, None)
+        (
+            rank,
+            name,
+            pool,
+            *others,
+            _,
+            value_locked,
+            apy1,
+            _,  # hourly removed for most cases it's 0.00 so it doesn't bring any value for user
+        ) = row_cleaned
+        auditors, collateral = collateral_auditors_parse(others)
+        auditors = ", ".join(aud.strip() for aud in auditors)
+        collateral = ", ".join(coll.strip() for coll in collateral)
+        results.append(
+            [
+                rank,
+                name,
+                pool,
+                auditors,
+                collateral,
+                value_locked,
+                apy1,
+            ]
+        )
+    df = pd.DataFrame(results, columns=columns).replace({"": None})
+    for col in ["return_year"]:
+        df[col] = df[col].apply(
+            lambda x: x.replace(" Yearly", "") if isinstance(x, str) else x
+        )
+    df["rank"] = df["rank"].astype(int)
+    df = wrap_text_in_df(df, w=30)
+    return df
+
+
+def get_top_volume_coins() -> pd.DataFrame:
+    """Scrapes top coins by trading volume "https://www.coingecko.com/en/coins/high_volume"
+
+    Returns
+    -------
+    pandas.DataFrame
+        rank, name, symbol, price, change_1h, change_24h, change_7d, volume_24h, market_cap
+    """
+    columns = [
+        COLUMNS["rank"],
```

Comment:
This doesn't add much value because rank is already rank, and this rank is pre-defined